### PR TITLE
Neo4j case study

### PR DESCRIPTION
**Pull request by VLazari**

### CODE REVIEW

**Comment by Anclagen** [comment]
I think what is here, looks good as far as I can see. I would say its needs some more content with regards to exactly what the graph data model is and how it works. I would also like some comparison/explanation of the graph database, vs No-SQL and SQL databases, with some indication of how they differ and what their best use cases are based on a projects requirements.

File: neo4j-case-study.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+-   **Reliable transactions:** Provided using ACID transactions, ensuring that database operations are completed successfully or not at all, making it a reliable solution for critical data.
+
+## Weaknesses
+
+-   **Higher resource requirements:** Compared to traditional relational databases, Neo4j requires higher resources for storage and processing, making it less suitable for use cases with limited resources.
+
+-   **Complex deployment:** Neo4j may require more complex deployment and maintenance compared to other databases, especially for large-scale deployments.
+
+-   **Analytics capabilities:** It may not be as well suited for more advanced analytics and data processing, compared to other databases.
```
Comment (Fermain) [inline neo4j-case-study.md:58]: Why not?

File: neo4j-case-study.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+-   **Document storage:** It is well suited for managing complex relationships between data but, it may not be as well suited for document storage, compared to other databases.
```
Comment (Fermain) [inline neo4j-case-study.md:60]: Any specific type of database that is good with documents?

**Review by Fermain — CHANGES_REQUESTED** [review]
This is a good submission on an interesting topic, I've prompted for a bit more detail in some cases but the spelling, grammar, tone and phrasing is absolutely fine.

I do feel like there is something missing however, which is contextualisation. Could you give an example of an application that requires this kind of datastore? You mention a few in the introduction, but why does a recommendation system require more than SQL? There's not much about the specific shortcomings of SQL that necessitate this tool.

Lastly, it would be nice to see a snippet of a Cypher query syntax.

File: neo4j-case-study.md (hunk 2 above)
Comment (VLazari) [reply neo4j-case-study.md]: I have add some more specifications and compared neo4j with non graph databases.

File: neo4j-case-study.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
+b---
```
Comment (Fermain) [inline neo4j-case-study.md:1]:
```suggestion
---
```

**Review by Fermain — APPROVED** [review]
This is a detailed and well structured submission on an interesting subject. You provide a lot of information in a small space by choosing words carefully - something your readers will appreciate.

One area for improvement would be tying this in with life as a developer. In what contexts might we want to use a graph style dbms? Perhaps there are particular applications in computing that go hand in hand with this style ;)

**Pull request merged**